### PR TITLE
udev: introduce symlink_atomic_full() and use it

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/basic/fs-util.c [hunk 1]
```diff
@@ -432,13 +432,21 @@ int symlink_idempotent(const char *from, const char *to, bool make_relative) {
...
-        _cleanup_free_ char *t = NULL;
+int symlink_atomic_full(const char *from, const char *to, bool make_relative) {
+        _cleanup_free_ char *relpath = NULL, *t = NULL;
         int r;
 
         assert(from);
         assert(to);
 
+        if (make_relative) {
+                r = path_make_relative_parent(to, from, &relpath);
```
Comment (daandemeyer) [inline src/basic/fs-util.c:443]: Why use `path_make_relative_parent()` here instead of `path_make_relative()`? I feel like this deserves a comment

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]
LGTM, but I don't quite follow why we need `path_make_relative_parent()` for relative symlinks

File: src/basic/fs-util.c (hunk 1 above)
Comment (yuwata) [reply src/basic/fs-util.c:443]: E.g. to: /path/to/aaa and from: /path/to/bbb, then
- path_make_relative(from, to) -> "../aaa"
- path_make_relative_parent(from, to) -> "aaa"

I will comment that somewhere in another PR.

File: src/udev/udev-node.c [hunk 2]
```diff
@@ -14,13 +14,15 @@
...
 #include "hexdecoct.h"
+#include "label.h"
 #include "mkdir-label.h"
 #include "parse-util.h"
 #include "path-util.h"
 #include "selinux-util.h"
 #include "smack-util.h"
 #include "stat-util.h"
 #include "string-util.h"
+#include "tmpfile-util.h"
```
Comment (daandemeyer) [inline src/udev/udev-node.c]: I don't think we need this include here now?

**Review by DaanDeMeyer — APPROVED** [review]
One thing, but LGTM

File: src/udev/udev-node.c (hunk 2 above)
Comment (yuwata) [reply src/udev/udev-node.c]: Indeed. Will drop.

**Comment by yuwata** [comment]
@DaanDeMeyer Thank you for reviewing many PRs! The above comments are addressed, upgrading the green label.

**Pull request merged**